torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -120,6 +145,192 @@ class SubgraphSlicer {
+        local_map[subgraph->outputs().at(i)] = subgraphNode->outputs().at(i);
+        node_inputs.erase(subgraph->outputs().at(i));
+      } 
+    }
+
+    WithInsertPoint wip(subgraphNode->next());
+
+    // these node inputs need to be added to subgraph's outputs
+    // put them in vmap
+    for (auto ni : node_inputs) {
+
+      if (local_map.count(ni) != 0) {
+        // this could happen if `n` uses two or more outputs
+        // of a constant node and we already clone the constant
+        // into the outer graph and mapped its outputs
+        continue;
+      }
+
+      Value* sno = nullptr;
+      if (ni->node()->kind() == prim::Constant) {
+        auto copy = subgraphNode->owningGraph()->createClone(ni->node(), env);
+        subgraphNode->owningGraph()->insertNode(copy);
+        // in case we have a multi-output const, map the rest of the outputs
+        // so we get to clone `n`, `n`'s clone will use the outputs of the 
+        // constant clone
+        for (auto i : c10::irange(n->outputs().size())) {
+          GRAPH_DEBUG("Mapping %", ni->node()->output(i)->debugName(), " to %", copy->output(i)->debugName());
+          local_map[ni->node()->output(i)] = copy->output(i);
+        }
+      }
+      else {
+        subgraph->registerOutput(ni);
+        sno = subgraphNode->addOutput();
+        sno->setType(ni->type());
+        GRAPH_DEBUG("Mapping %", ni->debugName(), " to %", sno->debugName());
+        local_map[ni] = sno;
+      }
+    }
+
+    auto copy = subgraphNode->owningGraph()->createClone(n, env);
+    GRAPH_DEBUG("copy ", *copy);
+
+    for (auto i : c10::irange(n->outputs().size())) {
+      auto oo = n->outputs()[i];
+      auto no = copy->outputs()[i];
+      no->copyMetadata(oo);
+      GRAPH_DEBUG("Mapping %", oo->debugName(), " to %", no->debugName());
+      local_map[oo] = no;    
+    }
+
+    subgraphNode->owningGraph()->insertNode(copy);
+
+    for (auto it = output_indices.rbegin(); it != output_indices.rend(); it++) {
+      auto replace_val = local_map[subgraph->outputs().at(*it)];
+      subgraphNode->outputs().at(*it)->replaceAllUsesWith(replace_val);
+      subgraphNode->eraseOutput(*it);
+      subgraph->eraseOutput(*it);
+    }
+
+    // size_t num_outputs = subgraph->outputs().size();
+    // for (int64_t i = num_outputs - 1; i >= 0; i++) {
+    //   subgraphNode->eraseOutput(i);
+    //   subgraph->eraseOutput(i);
+    // }
+
+    n->destroy();
+  }
+
+  std::vector<std::set<Value*, topo_cmp_value>> buildAliasedSets(std::shared_ptr<Graph> subgraph) {
+    auto outputs = subgraph->outputs();
+    AliasDb alias_db(subgraph);
+    TORCH_INTERNAL_ASSERT(outputs.size() > 1);
+    std::vector<std::set<Value*, topo_cmp_value>> res;
+    for (auto o : outputs) {
+      auto grouped = false;
+      for (auto& s : res) {
+        auto os = *s.begin();
+        GRAPH_DEBUG("comparing %", o->debugName(), " with %", os->debugName(), " result ", (alias_db.mayContainAlias(os, o)));
```

Comment:
we shouldn't be running `alias_db.mayContainAlias(os, o))` within the debug, we're effectively running it twice now